### PR TITLE
update all the things in preparation for v3.0.0 changes

### PR DESCRIPTION
**Pull request by boneskull**
- changed some opencollective-related copy; cc @xdamman
- generate sponsors and backers list in a Liquid loop
- use Kramdown to hide avatars in a section until they have all loaded, then fade into view quickly
- fix potential `http`/`https` asset disagreements
- update excluded files
- use `markdown-toc` package instead of a decrepit hack
- remove `Makefile`; use `p-s` which should help with portability
- update `package.json` info with license, contributors, original author
- update & simplify `README.md`

### CODE REVIEW

File: _includes/backers.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+## Backers
+
+Find Mocha helpful?  Become a [backer](https://opencollective.com/mochajs#support) and support Mocha with a monthly donation.
```
Comment (dasilvacontin) [inline _includes/backers.md:3]: Double space before 'Become'

File: README.md [hunk 2]
```diff
@@ -1,49 +1,32 @@
...
-For more information, refer to the [Jekyll Docs](http://jekyllrb.com/docs/home/) and [GitHub's Tutorial](https://help.github.com/articles/using-jekyll-with-pages/) on the subject.
-
-## Notes
-
-- To update `index.md` with the TOC, execute: `node_modules/.bin/toc-idx -i index.md --max-depth 1 --bullet "\- " index.md`
-- `bundle exec jekyll build` rebuilds the site without updating the TOC
-- The `_site` directory is where the generated site lives.  It is *not* under version control, because GitHub Pages generates it for us.
+## License
 
+:copyright: 2016 mochajs.org contributors.  Content licensed CC-BY-SA-4.0; code licensed MIT.
```
Comment (dasilvacontin) [inline README.md:32]: Double space before 'Content'

File: _includes/backers.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+## Backers
+
+Find Mocha helpful?  Become a [backer](https://opencollective.com/mochajs#support) and support Mocha with a monthly donation. 
+
+{: id="_backers" }
+{% for i in (0..29) %}[![](//opencollective.com/mochajs/backer/{{ i }}/avatar){: onload="window.avatars.backerLoaded()" }](https://opencollective.com/mochajs/backer/{{ i }}/website){: target="_blank"}{% endfor %}
```
Comment (dasilvacontin) [inline _includes/backers.md:6]: Neato.

**Comment by dasilvacontin** [comment]
Is there a reason for the double spaces? I've seen quite a few around.

**Comment by dasilvacontin** [comment]
It feels as if the live version has a bigger margin between the backers' avatars. It wouldn't hurt increasing them.

**Comment by dasilvacontin** [comment]
I'd say it looks better with a lighter black for the text, like `#717171` – one that doesn't cry and hurt for attention so much. Specially important at the top of the page, because the bolder description now brings more attention than the logo / motto themselves.

File: scripts/toc.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+const fs = require('fs');
+const path = require('path');
+const filepath = path.join(__dirname, '..', 'index.md');
+
+console.log('Updating TOC...');
+
+const input = fs.createReadStream(filepath);
+
+input.pipe(
+    utils.concat(input => fs.writeFileSync(filepath, toc.insert(String(input), {
```
Comment (dasilvacontin) [inline scripts/toc.js:21]: Would be nice to separate all this code into different statements / lines, which will be easier to understand thanks to variable names.

**Comment by dasilvacontin** [comment]
Great changes overall! :+1:

File: _includes/backers.md (hunk 1 above)
Comment (boneskull) [inline _includes/backers.md:3]: this is a habit I picked up in high school.  think it was APA style, or something.  heated debate out there regarding this, actually.   

but: the only way to get two consecutive spaces in HTML is to use `&nbsp;`, so it doesn't really matter here.

**Comment by boneskull** [comment]
> It feels as if the live version has a bigger margin between the backers' avatars. It wouldn't hurt increasing them.

I'll look into this.  a change here was not intentional.

**Comment by boneskull** [comment]
> Is there a reason for the double spaces? I've seen quite a few around.

yes, you can google "two spaces after end of sentence" or something similar, and read all the  arguments one way or another, if you are curious.  but my point remains: ` `is equivalent to`  `is equivalent to`             ` in html _anyway_, so it doesn't matter.

**UPDATE** you can't even have two consecutive spaces in GFM, so it doubly doesn't matter.

**Comment by boneskull** [comment]
> I'd say it looks better with a lighter black for the text, like #717171 – one that doesn't cry and hurt for attention so much. Specially important at the top of the page, because the bolder description now brings more attention than the logo / motto themselves.

I'll change it.  it appears you're looking at #44 as well, so I'll probably end up closing this PR and keeping that one.

**Comment by boneskull** [comment]
closing in lieu of #44

**Pull request closed without merging**